alea/include/alps/alea/internal/joined.hpp

```diff
@@ -18,12 +18,49 @@ namespace alps { namespace alea {
 
     struct circular_var;
     struct elliptic_var;
+
 }}
 
+namespace alps { namespace alea { namespace internal {
```

Comment:
Minor point: is `internal::` the same meaning as `detail::`? Should we use `detail::` for uniformity?
Or is it to follow some Eigen convention?

Basically, as far as i can understand, `detail::` (e.g., in Boost) means "implementation details, can be changed or disappear without warning, do not use!" (so to say, a "`private:` namespace"); while `internal::` (in Eigen) is expandable and Eigen [endorses it's use](https://eigen.tuxfamily.org/dox/TopicNewExpressionType.html) (so to say, a "`protected:` namespace"). 

Which one is the case here? Are non-alea developers allowed to add classes to `alps::alea::internal`? Why would they want to do it?

Reply:
It is a little bit of both, but closer to boost's `detail`, I think.  I would put it like this: it can change as long as the change does not break any real-world code :-)

I don't particularly like the name `detail` for a namespace, because it suggests that this is just a detail rather than a protected/private, i.e., internal, part of the interface.